packages/gatsby/src/internal-plugins/functions/gatsby-node.ts

```diff
@@ -70,9 +72,25 @@ async function ensureFunctionIsCompiled(
 const createGlobArray = (siteDirectoryPath, plugins): Array<IGlobPattern> => {
   const globs: Array<IGlobPattern> = []
 
+  function globIgnorePatterns(
+    root: string,
+    pluginName?: string
+  ): Array<string> {
+    const nestedFolder = pluginName ? `/${pluginName}/**/` : `/**/`
+
+    return [
```

Comment:
I know that for `src/pages` we ignore more:

https://github.com/gatsbyjs/gatsby/blob/d5e69aefcb86d3c581dd4e1192362e63efce3a2e/packages/gatsby-page-utils/src/validate-path.ts#L4-L30

But underscore, `template-` could be a breaking change. JSON and YAML doesn't make much sense to me